src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Windows.targets

```
@@ -70,6 +70,16 @@ Copyright (c) .NET Foundation. All rights reserved.
     <FrameworkReference Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '8.0')) and '$(TargetPlatformIdentifier)' == 'Windows' and '$(UseUwp)' == 'true' " Include="Microsoft.Windows.SDK.NET.Ref.Xaml" />
   </ItemGroup>
 
+  <!-- If WindowsSdkPackageVersion is specified and it is an older version before profile support was added, 
+       add the non profile framework reference. -->
+  <ItemGroup Condition=" '$(IncludeWindowsSDKRefFrameworkReferences)' == 'true'
+                        and '$(WindowsSdkPackageVersion)' != ''
+                        and '$(UseUwp)' != 'true'
+                        and $([System.Version]::Parse('$(WindowsSdkPackageVersion.Split('-')[0])').Build) &lt;= 26100
+                        and $([System.Version]::Parse('$(WindowsSdkPackageVersion.Split('-')[0])').Revision) &lt;= 34">
```

I'm not sure if I got the exact version boundary right, but here's a better way to do version comparison in MSBuild:

```suggestion
                        and $([MSBuild]::VersionLessThan('$(WindowsSdkPackageVersion)', '10.0.26100.34'))>
```

---

I thought that check was intentional? Eg. if someone were using '10.0.17763.45', that would be correct, but wouldn't that check still consider that as a lower version, which would be wrong in that specific case? 🤔

---

Yes, that what was reason why both were split up.  I am checking that it is an OS version that we have shipped so far and is an older version of that package for that OS version.